oslabs-beta/DockerLocal
# Error handling

## Description:
- Most controllers in the codebase did not have error handling, and this produced no errors when errors occurred. This makes it more difficult to understand where the errors are occurring and to find a solution.
- Codebase also had lots of TSLint errors. I mostly added return types and fixed whitespace issues.
## Changes I Made:
- **`src/client/components/projects/ProjectPage.tsx`**: format according to eslint
- **`src/client/helpers/cookieClientHelper.ts`**: Add handling for a empty cookie so error will not appear on front-end
- **`src/scripts/findDockerfiles.sh`**: Add error handling if no Dockerfile is found in the active Project. It is not set up to trigger the global error handler. Maybe in the future we can pass my error message directly to the user so they can add a repo with a dockerfile.
- **`src/scripts/sshKeygen.sh`**: Remove temporary folders before getting the SSH Key and creating temporary folders to ensure that there are no folders being stored on the user's computers.
- **`src/server/config/passport-setup.ts`**: Add spaces in front of comments according to eslint
- **`src/server/controllers/apiController.ts`**: File was no longer being used and was deleted.
- **`src/server/controllers/authController.ts:`** add error handling to saveAcessToken, getNameAndTokenFromCookies, saveUSerInforAndRepos, fix comment spacing, add types, change import statement.
- **`src/server/controllers/configController.ts:`** change import statements according to tslint, make read and write file synchronous to work with import statement,  add error handling to each fs method.
- **`src/server/controllers/dockerController.ts`**: add spacing for easier readability. Add error handling to all functions and fs methods. Add specific error handling if active project has no repo with a dockerfile.
- **`src/server/controllers/gitController.ts`**:Add types and error handling
- **`src/server/controllers/helpers/shellHelper.ts`**: Add descriptions to functions and return type
- **`api-route.ts`**: Add return types and error handling
- **`auth-route.ts`**: rewrite two import statements according to tslint. Add spaces to comments
- **`config-route.ts`**: Remove trailing whitespaces
- **`docker-route.ts`**: Add descriptive comments
- **`src/server/server.ts`**: change import statements according to tslint.

## How to Test:
- npm install
- npm start
- Check the Devtools. No error pops up about replace of undefined.
- Login. 
- Check all functionality.
  - Create Project
  - Add Repos
  - Clone Repos
  - Create Dockerfile
- Create a dockerfile with a project that has only cloned repos without dockerfiles- you should see error: cannot compose file in the modal.
